main.go

```diff
@@ -280,12 +281,21 @@ func getBotChannels(api *slack.Client) ([]Channel, error) {
 	return results, nil
 }
 
-func isUpdateInChannelHistory(update StatusPageIncidentUpdate, api *slack.Client, channel Channel) (bool, error) {
+func isUpdateInChannelHistory(update StatusPageIncidentUpdate, api *slack.Client, channel Channel, retry int) (bool, error) {
 	historyResp, err := api.GetConversationHistory(&slack.GetConversationHistoryParameters{
 		ChannelID: channel.ID,
 	})
 	if err != nil {
 		log.Printf("Error getting conversation history: %v", err)
+		// If the error is because we hit a rate limit, sleep for the given time and try again
+		switch err.(type) {
+		case *slack.RateLimitedError:
+			if retry < retryMax {
+				time.Sleep(err.(*slack.RateLimitedError).RetryAfter)
```

Comment:
@jradtilbrook This ".RetryAfter" is new to me - what guarantees do we have of it being there?

Reply:
It's on the RateLimitedError type from the slack api. It will be here because I'm first using the type switch to confirm the `err` is of that type which has that property available. (I still just need to cast it to access it though).